lib/gollum-lib/filter/tags.rb

```diff
@@ -73,21 +74,31 @@ def is_preformatted?(node)
   #
   # Returns the String HTML version of the tag.
   def process_tag(tag)
-    if tag =~ /^_TOC_/
+    link_part, extra = parse_tag_parts(tag)
+    mime = MIME::Types.type_for(::File.extname(link_part.to_s)).first
+
+    result = if link_part =~ /^_TOC_/
       %{[[#{tag}]]}
-    elsif tag =~ /^_$/
+    elsif link_part =~ /^_$/
       %{<div class="clearfloats"></div>}
-    elsif (html = process_include_tag(tag))
-      html
-    elsif (html = process_image_tag(tag))
-      html
-    elsif (html = process_external_link_tag(tag))
-      html
-    elsif (html = process_file_link_tag(tag))
-      html
-    else
-      process_page_link_tag(tag)
+    elsif link_part =~ /^include:/
+      process_include_tag(link_part)
+    elsif mime && mime.content_type =~ /^image/
+      process_image_tag(link_part, extra)
+    elsif external = process_external_link_tag(extra, link_part)
+      external
     end
+    result ? result : process_link_tag(link_part, extra)
+  end
+
+  def process_link_tag(link_part, extra)
+    process_file_link_tag(link_part, extra) || process_page_link_tag(link_part, extra)
```

Comment:
Yes, that's what bothered me originally. I would like to just walk through the repo searching for the requested paths, returning `Page` objects if the found object has a valid page extensions and `File` objects otherwise. Then `process_link_tag` should choose what to with each link based on what kind of object it gets back (if any).